docs/user_guide/mitigation.rst

```diff
@@ -74,6 +74,109 @@ Preprocessing
+        -\mathbf{x}_{i}\right\|^{2} \\
+        \text{subject to} \\
+        \frac{1}{n} \sum_{i=1}^{n} \mathbf{z}_{i}\left(\mathbf{s}_{i}-\overline{\mathbf{s}}
+        \right)^{T}=\mathbf{0}
+
+The solution to this problem is found by centering sensitive features, fitting a
+linear regression model to the non-sensitive features and reporting the residual.
+The columns in :math:`S` will be dropped from the dataset :math:`X` but the
+amount of correlation that is removed can be controlled using the
+:code:`alpha` parameter. This is described as follows:
+
+.. math::
+        X_{\text{tfm}} = \alpha X_{\text{filtered}} + (1-\alpha) X_{\text{orig}}
+
+Note that the lack of correlation does not imply anything about statistical dependence.
+Therefore, we expect this to be most appropriate as a preprocessing step for
+(generalized) linear models.
+
+In the example below, a subset of the `adult dataset <https://www.openml.org/d/1590>`_
+is loaded and the correlation between sex and the non-sensitive features is removed.
+This procedure will drop the sensitive features from the dataset.
+
+.. doctest:: mitigation
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> from fairlearn.preprocessing import CorrelationRemover
+    >>> import pandas as pd
+    >>> from sklearn.datasets import fetch_openml
+    >>> data = fetch_openml(data_id=1590, as_frame=True)
+    >>> X = data.data[['age', 'fnlwgt', 'education-num', 'sex']]
+    >>> X = pd.get_dummies(X)
+    >>> X = X.drop('sex_Male', axis=1)
+    >>> cr = CorrelationRemover(sensitive_feature_ids=['sex_Female'])
+    >>> cr.fit(X)
+    CorrelationRemover(sensitive_feature_ids=['sex_Female'])
+    >>> X_transform = cr.transform(X)
+
+In the visualization below, we see the correlation values in the
+original dataset. We are particularly interested in the correlations
+between the 'sex_Female' column and the three non-sensitive attributes
+'age', 'fnlwgt' and 'education-num'. The target variable is also
+included in these visualization for completeness, and it is defined as
+a binary feature which indicated an income higher than $50k. We see that
+'sex_Female' is not highly correlated with the three mentioned attributes,
+but we want to remove these correlations nonetheless.
+The code for generating the correlation matrix can be found in
+`this example notebook
+<../auto_examples/plot_correlationremover_before_after.html>`_.
+
+.. figure:: ../auto_examples/images/sphx_glr_plot_correlationremover_before_after_001.png
+    :align: center
+    :target: ../auto_examples/plot_correlationremover_before_after.html
+
+In order to see the effect of :class:`CorrelationRemover`, we visualize
+how the correlation matrix has changed after the transformation of the
+dataset. Due to rounding, some of the 0.0 values appear as -0.0. Either
+way, the :code:`CorrelationRemover` successfully removed all correlation
+between 'sex_Female' and the other columns while retaining the correlation
+between the other features.
```

Comment:
Maybe note that the correlation of age with the target variable has decreased a bit as well.

Reply:
Though we might want to wait with such details after we have replaced the Adult dataset in this example.

Reply:
That's what I thought as well :). I tried to keep the text as independent from the dataset as possible, although that is not always possible ofcourse.